indexer/e2e_tests/setup.go

```diff
@@ -105,14 +118,54 @@ func createE2ETestSuite(t *testing.T) E2ETestSuite {
 	require.NoError(t, err)
 
 	indexerCtx, indexerStop := context.WithCancel(context.Background())
-	t.Cleanup(func() { indexerStop() })
 	go func() {
 		err := indexer.Run(indexerCtx)
-		require.NoError(t, err)
+		if err != nil { // panicking here ensures that the test will exit
+			// during service failure. Using t.Fail() wouldn't be caught
+			// until all awaiting routines finish which would never happen.
+			panic(err)
+		}
 	}()
 
+	apiLog := testlog.Logger(t, log.LvlInfo).New("role", "indexer_api")
+
+	apiCfg := config.ServerConfig{
+		Host: "127.0.0.1",
+		Port: 0,
+	}
+
+	mCfg := config.ServerConfig{
+		Host: "127.0.0.1",
+		Port: 0,
+	}
+
+	api := api.NewApi(apiLog, db.BridgeTransfers, apiCfg, mCfg)
+	apiCtx, apiStop := context.WithCancel(context.Background())
+	go func() {
+		err := api.Run(apiCtx)
+		if err != nil {
+			panic(err)
+		}
+	}()
+
+	t.Cleanup(func() {
+		apiStop()
+		indexerStop()
+	})
+
+	// Wait for the API to start listening
+	time.Sleep(1 * time.Second)
```

Comment:
ffti: can also wait for the health check endpoint to return 200

Reply:
That is true but the port would be unknown for some duration of time which is why we're doing this wait here